core/shared/src/main/scala/zio/ZSchedule.scala

```diff
@@ -713,14 +811,60 @@ object ZSchedule {
   final def logInput[R, A](f: A => ZIO[R, Nothing, Unit]): ZSchedule[R, A, A] =
     identity[A].logInput(f)
 
+  /**
+   * A schedule that waits forever when updating or initializing.
+   */
+  final val never: Schedule[Any, Nothing] =
+    ZSchedule[Any, Nothing, Any, Nothing](UIO.never, (_, _) => UIO.never, (_, never) => never)
+
+  /**
+   * A schedule that executes once.
+   */
+  final val once: Schedule[Any, Unit] = recurs(1).unit
+
+  /**
+   * A schedule that sleeps for random duration that is uniformly distributed in the given range.
+   * The schedules output is the duration it has slept on the last update, or 0 if it hasn't updated yet.
+   */
+  final def randomDelay(min: Duration, max: Duration): ZSchedule[Random with Clock, Any, Duration] = {
+    val minNanos = min.toNanos
+    val maxNanos = max.toNanos
+    ZSchedule[Clock with Random, Duration, Any, Duration](
+      ZIO.succeed(Duration.Zero), {
+        case _ =>
+          random.nextLong(maxNanos - minNanos + 1).flatMap { n =>
+            val duration = Duration.fromNanos(n - minNanos)
+            clock.sleep(duration).as(duration)
+          }
+      },
+      { case (_, duration) => duration }
+    )
+  }
+
+  /**
+   * A schedule that sleeps for random duration that is normally distributed.
+   * The schedules output is the duration it has slept on the last update, or 0 if it hasn't updated yet.
+   */
+  final def randomDelayNormal(mean: Duration, std: Duration): ZSchedule[Random with Clock, Any, Duration] =
+    ZSchedule[Clock with Random, Duration, Any, Duration](
+      ZIO.succeed(Duration.Zero), {
+        case _ =>
+          random.nextGaussian.flatMap { n =>
+            val duration = mean + std * n
+            clock.sleep(duration).as(duration)
+          }
+      },
+      { case (_, duration) => duration }
+    )
+
   /**
    * A schedule that recurs the specified number of times. Returns the number
    * of repetitions so far.
    *
    * If 0 or negative numbers are given, the operation is not done at all so
    * that in `(op: IO[E, A]).repeat(Schedule.recurs(0)) `, op is not done at all.
    */
-  final def recurs(n: Int): Schedule[Any, Int] = forever.whileOutput(_ <= n)
+  final def recurs(n: Int): Schedule[Any, Int] = forever.whileOutput(_ < n)
```

Comment:
Intentional change?

Reply:
Sadly check subtly changed its semantics as we now have to extract before updating and not after.
This is a consequence of that change